scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
+* Are the warnings OK? Do I need an option to disable them?
+* Is 100% pep8 compliance necessary?
+* How should unit tests be performed on both simplex and ip?
```

Comment:
I can define all the tests as class methods, but what other changes need to be made in order to actually run the tests when I execute `test_linprog_ip.py`?  It's sort of magic to me that all the functions defined in `test_linprog_ip.py` are automatically run (as tests) when `test_linprog_ip.py` is executed as a script, so I don't understand how the tests are to be run when they are class methods. Does this make sense?